api/Api.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * This file is part of the ArcTouch CodeChallenge requested during the Interview Process.
+ *
+ * @package CodeChallenge
+ * @author Matheus Muller Borges
+ * @version 0.0.1
+ */
+namespace CodeChallenge;
+
+require_once dirname(__DIR__).'/bootstrap.php';
+
+
+/**
+ * Class Api
+ * @package CodeChallenge
+ */
+class Api {
+	/** @var  Router */
+	private $router;
+
+	/**
+	 * @var \Tmdb\Client
+	 */
+	private $client;
+
+	/**
+	 * Holds the TMDb images base url from the TMDB Configuration API.
+	 * @var string
+	 */
+	private $imagesBaseUrl;
+
+
+	public function __construct() {
+		$token = new \Tmdb\ApiToken('1f54bd990f1cdfb230adb312546d765d');
+		$this->client = new \Tmdb\Client($token, [ 'secure' => false ]);
+
+
+		$this->loadImagesBaseUrl();
+		$this->buildRoutes();
+	}
+
+	/**
+	 * Build the routes for the API
+	 */
+	private function buildRoutes() {
+		$this->router = new Router(function() {
+			return ['status_message' => 'The resource you requested could not be found'];
+		});
+
+		$this->router->addRoute('/v1/movie/(.*)',  [$this, 'getMovie']);
+		$this->router->addRoute('/v1/upcoming/(.*)',  [$this, 'getUpcoming']);
+
+		$this->router->addRoute('/v1/search/:query', [$this, 'searchMovie']);
+	}
+
+	/**
+	 * Returns the TMBb movie data for the given movie id
+	 * @param int $id TMDb movie id
+	 *
+	 * @return array The movie data
+	 */
+	public function getMovie($id) {
+		try {
+			$movie = $this->client->getMoviesApi()->getMovie($id);
+			return $movie;
+		} catch(\Exception $e) {
+			return ['status' => $e->getMessage()];
+		}
+	}
+
+	/**
+	 * Get TMDb upcoming movies list
```

Comment:
What do you think about the comments on every method and class. Shouldn't the method name or class name be sufficient to describe it's purpose? I mean, I feel like I'm in 2000 programming Java using eclipse IDE :)

Reply:
Yes, I do agree that adding obvious comments may be too cumbersome. The method's name and its arguments should be clear enough to describe it's purpose. However, if the client requires the code documentation, it can be easily generated using PHPDoc. Also, a documented code allows the IDE hinting.